### PR TITLE
Adding exception handling for ServerNotAvailableException

### LINKED ISSUE #17192: Distinguish auth backend unavailability from of auth failure
When the LDAP backend is unavailable (for instance due to maintenance) attempted user logins result in replies that indicate authentication failure. As far as we know it's impossible to distinguish these from authentication failures caused by invalid credentials.

When receiving this message, the desktop client believes that the credentials are invalid and asks users for the correct password (see owncloud/client#2848). That's not what we want.

If the client instead got the message that no answer can be produced because the auth backend is temporarily unavailable, it could assume the credentials are still valid and try again later, without bothering the user about it.

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@ckamm @LukasReschke @PVince81 

fixes #17192

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
This does not fix #17192 - but returns a 503 instead of a 500 during auth.

**Comment by nickvergessen** [comment]
Clean diff: https://github.com/owncloud/core/pull/17224/files?w=1

**Comment by DeepDiver1975** [comment]
@blizzz please retest - THX

**Comment by MorrisJobke** [comment]
@DeepDiver1975 I pushed also the latest commit to the jenkins PR ;)

**Comment by MorrisJobke** [comment]
@blizzz please test :)

File: lib/private/connector/sabre/auth.php [hunk 1]
```diff
@@ -30,7 +30,13 @@
  */
 namespace OC\Connector\Sabre;
 
-class Auth extends \Sabre\DAV\Auth\Backend\AbstractBasic {
+use Exception;
+use Sabre\DAV\Auth\Backend\AbstractBasic;
+use Sabre\DAV\Exception\NotAuthenticated;
+use Sabre\DAV\Exception\ServiceUnavailable;
+use Sabre\DAV\Server;
```
Comment (PVince81) [inline lib/private/connector/sabre/auth.php]: This will :bomb: again due to namespace conflict.

Comment (DeepDiver1975) [inline lib/private/connector/sabre/auth.php]: I'll update this asap

**Comment by blizzz** [comment]
That's better now, getting the 503 exception :+1:

**Comment by DeepDiver1975** [comment]
Another review please? @MorrisJobke @PVince81 @LukasReschke @ckamm THX

**Comment by PVince81** [comment]
Code looks good, webdav still works :+1:

**Comment by MorrisJobke** [comment]
@PVince81 @DeepDiver1975 We need to backport this now to stable8.1

**Comment by MorrisJobke** [comment]
> @PVince81 @DeepDiver1975 We need to backport this now to stable8.1

#17266

**Comment by DeepDiver1975** [comment]
THX @MorrisJobke

**Pull request merged**